### DISCUSSION TITLE
CD database errors and odd display in 20240122 (and earlier)

### DISCUSSION THREAD

**Discussion by Vseason** [rern/rAudio#1581] [Bug Reports]
<i>(Migrated from a comment in the general section, since it is more appropriately a bug report…)</i>

@rern Did the online CD database entirely stop working at some point?

Early on, whenever I loaded an audio CD in the drive, it would successfully retrieve the track listing and cover artwork. But now, even testing against discs that were previously recognized (and I photographed the screen while playing a few of them, to show a friend, so I’m absolutely certain of checking the same disc!) the message is always “ID of CD not found in database.”

Additionally, rather than simply leaving the titles empty since the disc is unidentified, or saying something like “Unknown,” it now shows “•&ensp&ZeroWidthSpace;;•&ensp&ZeroWidthSpace;;•” in their place.

Although CDs do still play, it is not currently possible to switch to the active playlist whenever a CD is playing, and the currently playing track numbers are not updated on the HDMI display unless playback is paused and restarted.

**Comment by Vseason** [comment]
This issue is still present, as described, in 20240127.

**Reply by rern** [reply]
The CD database server `gnudb.org` has been down for quite a while.

**Reply by Vseason** [reply]
Ah! Is there any chance the CD lookup could be switched to musicbrainz.org instead, in order to restore disc identification and individual track selection?

**Comment by Vseason** [comment]
In 20240219, CD track selection is restored (hooray, and thank you!), although oddly the track number and playback time do not update after the first track concludes, even though playback successfully continues.

The CD display oddities remain as previously described:
![IMG_6768](https://github.com/rern/rAudio/assets/107371732/6df8d852-ce9b-4256-b7f3-c446a24e6a8c)

**Reply by rern** [reply]
Has the latest update got it back to normal?

**Reply by Vseason** [reply]
I’ll check after work tonight! (Sorry for the delay in noticing the new update. It’s been a busy week here.)

**Reply by Vseason** [reply]
@rern This does indeed seem to have fixed the visual oddity unexpectedly of seeing HTML code for an en-space. Thank you!

Additionally, track transitions and elapsed time per track are properly registering again. Hooray!

Restoring disc ID retrieval (from an alternate, functional database?) would be wonderful at some point in the future, but in terms of functional CD playback the latest update appears to have done the trick, and I am deeply appreciative.

EDIT: Oddly, the “autoplay” option for CDs seems to have disappeared. I’ve not yet poked around enough to see whether that was just relocated, though.

EDIT2: … and hmm, after several tracks, the local HDMI display lost count again. (The web view on my iPhone still showed the correct track and time info.)
![IMG_6830](https://github.com/rern/rAudio/assets/107371732/3c305f0f-e3db-4cd5-930a-6fde75b440dc)
